AztecProtocol/aztec-v1
# Laurence's Huff changes & ERC20 implementation

## Description

1. Added missing `callvalue` argument in the benchmark test which previously stopped the whole thing from working, corrected usage documentation and fixed a minor typo.

2. Added ERC20 implementation in Huff with JavaScript interface.

3. Changed Huff `runtime.js` so that the VM is initialised in the JavaScript file calling the Huff macros, to allow contract state variables to persist between macro calls. Also edited existing test files to accommodate for this change.

## Testing instructions

Run `yarn benchmark` to demonstrate point 1.

Run `erc20_test.js` in the ERC20 package to see the ERC20 contract in action.

## Types of changes



Bug fix (non-breaking change which fixes an issue)

New feature (non-breaking change which adds functionality)



## Checklist:




Edited 17/07/19
